### DISCUSSION TITLE
how to store location points to database

### DISCUSSION THREAD

**Discussion by khaldon** [adonisjs/core#4679] [Help]
I use postgres. I want to store location point. accidentally I found `table.point()`. Is that dedicated to store points of location? if not Do I need to integrate postgis ?

**Comment by thetutlage** [comment]
Yeah, I think you will have to use the `knex-postgis` package for it and also you might have to read a bit about its usage on the internet. I am not super familiar with the Postgis extension.

Here's an article on integrating Postgis with AdonisJS https://www.mattstrayer.com/posts/adonis-v6-postgis/

**Comment by naveenbasyal** [comment]
> I use postgres. I want to store location point. accidentally I found `table.point()`. Is that dedicated to store points of location? if not Do I need to integrate postgis ?

Have you got any solution ??

**Reply by khaldon** [reply]
right now no

**Comment by PuneetSharma52** [comment]
Hey i also faced the same issue and I resolved it using:
1. I used string instead of point() so that I can pass the coordinates like "29.26544458668772, 76.04060242729946"
2. Then I split it using split function then I parsed it into two variables with the type integers,
3. Then I used raw postgis query on them to store them into the geometry type and is fully functional,
4. Attaching the code snippet I used,

### Code
const [latitude, longitude] = locationString
      .split(' ')
      .map((coord: string) => Number.parseFloat(coord))
    const location = await Location.create({
      id: uuid(),
      customerId,
      type,
    })
await location.save()
await db.rawQuery(
  `UPDATE locations SET location = ST_GeogFromText('SRID=4326;POINT(${longitude} ${latitude})')::geography WHERE customer_id = ?`,
  [customerId]
)
